File: spam.js
const { Builder, By } = require("selenium-webdriver");
const faker = require("faker");
const { execSync } = require("child_process");
const os = require("os");

// 🔥 Step 1: Auto-install Chrome & ChromeDriver (For Codespaces/VPS)
function installChrome() {
    try {
        console.log("🚀 Checking & Installing Chrome + ChromeDriver...");
        execSync("sudo apt update && sudo apt install -y wget curl unzip", { stdio: "inherit" });
        execSync("wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb", { stdio: "inherit" });
        execSync("sudo dpkg -i google-chrome-stable_current_amd64.deb || sudo apt --fix-broken install -y", { stdio: "inherit" });
        execSync("wget https://chromedriver.storage.googleapis.com/$(curl -sS https://chromedriver.storage.googleapis.com/LATEST_RELEASE)/chromedriver_linux64.zip", { stdio: "inherit" });
        execSync("unzip chromedriver_linux64.zip", { stdio: "inherit" });
        execSync("sudo mv chromedriver /usr/local/bin/", { stdio: "inherit" });
        execSync("sudo chmod +x /usr/local/bin/chromedriver", { stdio: "inherit" });
        console.log("✅ Chrome & ChromeDriver Installed Successfully!");
    } catch (err) {
        console.error("❌ Error installing Chrome/ChromeDriver:", err);
    }
}

// Install Chrome if missing (For Codespaces/VPS)
if (os.platform() !== "win32") {
    installChrome();
}

// 🔥 Step 2: Define Scammer Info & Telegram Support URL
const scammerUsername = "scammer123";  // Change to real scammer username
const scammerChannel = "https://t.me/scammerchannel";  // Change to real scammer channel
const telegramSupportURL = "https://telegram.org/support";

// 🔄 Step 3: Infinite Loop for Continuous Reporting
(async function spamReportLoop() {
    try {
        while (true) {
            let driver = await new Builder().forBrowser("chrome").build();

            try {
                console.log("🚀 Opening Telegram Support Page...");
                await driver.get(telegramSupportURL);

                // Generate Random Details
                let randomName = faker.name.findName();
                let randomEmail = faker.internet.email();
                let randomPhone = `+${faker.random.number({ min: 100, max: 999 })}${faker.random.number({ min: 1000000000, max: 9999999999 })}`;
                let scamDescription = `
This user (@${scammerUsername}) is running a **carding scam** and tricking people into sending money by promising illegal credit card details, bank logs, and other fraudulent services. Many victims have already been scammed.

He is using Telegram to promote his illegal activities and scam people under false promises. His Telegram channel is actively spreading fraudulent content and misleading users.

🚨 Scammer's Telegram Channel: ${scammerChannel}

Please take immediate action and ban this user and his channel before more people fall victim.
                `;

                console.log(`👤 Name: ${randomName}`);
                console.log(`📧 Email: ${randomEmail}`);
                console.log(`📱 Phone: ${randomPhone}`);
                console.log(`🚨 Reporting: @${scammerUsername}`);

                // Fill Description
                await driver.findElement(By.name("message")).sendKeys(scamDescription);
                // Fill Name
                await driver.findElement(By.name("name")).sendKeys(randomName);
                // Fill Email
                await driver.findElement(By.name("email")).sendKeys(randomEmail);
                // Fill Phone
                await driver.findElement(By.name("phone")).sendKeys(randomPhone);
                // Click Submit
                await driver.findElement(By.css("button[type='submit']")).click();

                console.log("✅ Report Submitted!");

            } catch (err) {
                console.error("❌ Error:", err);
            }

            // Close the browser
            await driver.quit();

            // Wait before next report (to avoid detection)
            await new Promise(resolve => setTimeout(resolve, 10000)); // 10 seconds delay
        }
    } catch (err) {
        console.error("❌ Fatal Error:", err);
    }
})();

File: spam.py
import random
import string
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

# Telegram Support URL
support_url = "https://telegram.org/support"

# ⚠️ Enter scammer's details manually
scammer_username = "scammer123"  # Change this to the actual scammer's username
scammer_channel = "https://t.me/scammerchannel"  # Change this to the actual channel link

# Scam Description Function
def generate_scam_description():
    return f"""
This user (@{scammer_username}) is running a **carding scam** and tricking people into sending money by promising illegal credit card details, bank logs, and other fraudulent services. Many victims have already been scammed.

He is using Telegram to promote his illegal activities and scam people under false promises. His Telegram channel is actively spreading fraudulent content and misleading users.

🚨 Scammer's Telegram Channel: {scammer_channel}

Please take immediate action and ban this user and his channel before more people fall victim.
"""

# Function to generate a random name
def generate_random_name():
    first_names = ["John", "Mike", "Alice", "Sarah", "David", "Emma", "Robert", "Olivia"]
    last_names = ["Smith", "Johnson", "Brown", "Taylor", "Anderson", "Thomas", "Harris"]
    return f"{random.choice(first_names)} {random.choice(last_names)}"

# Function to generate a random email
def generate_random_email():
    domains = ["gmail.com", "yahoo.com", "outlook.com", "protonmail.com"]
    email_name = ''.join(random.choices(string.ascii_lowercase + string.digits, k=10))
    return f"{email_name}@{random.choice(domains)}"

# Function to generate a random phone number
def generate_random_phone():
    return f"+{random.randint(100, 999)}{random.randint(1000000000, 9999999999)}"

# Infinite Loop for Continuous Reporting
try:
    while True:
        # Generate new random values
        random_name = generate_random_name()
        random_email = generate_random_email()
        random_phone = generate_random_phone()
        scam_description = generate_scam_description()

        # Set up Selenium WebDriver
        driver = webdriver.Chrome()  # Ensure you have ChromeDriver installed
        driver.get(support_url)

        time.sleep(2)  # Let the page load

        try:
            # Step 1: Fill in the description
            desc_field = driver.find_element(By.NAME, "message")
            desc_field.send_keys(scam_description)

            # Step 2: Enter random full name
            name_field = driver.find_element(By.NAME, "name")
            name_field.send_keys(random_name)

            # Step 3: Enter random email address
            email_field = driver.find_element(By.NAME, "email")
            email_field.send_keys(random_email)

            # Step 4: Enter random phone number
            phone_field = driver.find_element(By.NAME, "phone")
            phone_field.send_keys(random_phone)

            # Step 5: Submit the form
            submit_button = driver.find_element(By.CSS_SELECTOR, "button[type='submit']")
            submit_button.click()

            print(f"✅ Report Submitted! \n👤 Name: {random_name}\n📧 Email: {random_email}\n📱 Phone: {random_phone}\n🚨 Reported Username: @{scammer_username}\n📢 Channel: {scammer_channel}")

        except Exception as e:
            print("❌ Error:", str(e))

        # Close the browser
        time.sleep(5)
        driver.quit()

        # Wait before next report (to avoid detection)
        time.sleep(10)  # Adjust this delay if needed

except KeyboardInterrupt:
    print("\n❌ Script Stopped by User!")
